src/core/agent/services/identifierService.ts

```diff
@@ -128,18 +127,33 @@ class IdentifierService extends AgentService {
     const operation = await this.signifyClient
       .identifiers()
       .create(signifyName); //, this.getCreateAidOptions());
-    await operation.op();
+    let op = await operation.op();
+    const signifyOpName = op.name;
     const addRoleOperation = await this.signifyClient
       .identifiers()
       .addEndRole(signifyName, "agent", this.signifyClient.agent!.pre);
     await addRoleOperation.op();
     const identifier = operation.serder.ked.i;
+    const isPending = !op.done;
+    if (isPending) {
+      op = await waitAndGetDoneOp(this.signifyClient, op, 2000);
```

Comment:
`2000` is the timer from the start of the function, not for here

Reply:
@iFergal Can you suggest how to check time after 2s? Because we need op.name to check.

Reply:
```
const startTime = Date.now();
...
  op = await waitAndGetDoneOp(this.signifyClient, op, 2000 - (Date.now() - startTime));
```

Reply:
pushed in https://github.com/cardano-foundation/cf-identity-wallet/pull/497/commits/5188be41bd47cdbff351e18d05094a10d8805d83